src/main/java/com/singh/astha/medicinereminder/services/DosageService.java

```diff
@@ -5,12 +5,13 @@
 import com.singh.astha.medicinereminder.dtos.ResponseDto.MedicineResponseDto;
 
 import javax.transaction.Transactional;
+import java.text.ParseException;
 import java.util.List;
 
 @Transactional
 public interface DosageService {
 
-    MedicineResponseDto updateDosage(Long userId, DosageHistoryRequestDto dosageHistoryRequestDto);
+    MedicineResponseDto updateDosage(Long userId, DosageHistoryRequestDto dosageHistoryRequestDto) throws ParseException;
```

Comment:
ParseException is a generic exception for all the Parse issues, so don't let leak through service layer.
Instead catch it inside the service layer and throw the ResponseException.